api/workAllocation2/index.ts

```diff
@@ -71,25 +75,29 @@ export async function getTask(req: EnhancedRequest, res: Response, next: NextFun
  */
 export async function searchCase(req: EnhancedRequest, res: Response, next: NextFunction) {
   try {
-    const searchRequest = req.body.searchRequest;
-    const view = searchRequest.search_by === 'judge' ? 'judicial' : 'caseworker';
-    const basePath = prepareSearchCaseUrl(baseWorkAllocationTaskUrl, `myCases?view=${view}`);
-    const postCasePath = preparePaginationUrl(req, basePath);
-
-    // TODO below call mock api will be replaced when real api is ready
-    const promise = await handlePost(postCasePath, searchRequest, req);
-
+    const view = req.body.view;
+    const roleAssignments: any[] = req.session.roleAssignmentResponse;
+    let promise;
```

Comment:
This is not promise. Because you already resolved the promise. Could you please call it response?
And also can you create a interface for the type you receiving the response for

Reply:
Replied in the channel the await.* is returning a promise and not realtime response.

For the 2nd comment the elastice search api not able to return a fixed payload and the returning data are dynamic.

Reply:
let promise: AxiosResponse;